lib/actions/fulfilSubOrder.js

```diff
@@ -0,0 +1,106 @@
+"use strict";
+const request        = require('request');
+const requestOptions = require('../requestOptions.js');
+const handleResponse = require('../handleResponse.js');
+
+exports.process = processAction;
+
+function processAction(message, config) {
+  let self = this;
+
+  // submits the HTTP request to the VMS to fulfil the order
+  function fulfilSubOrder(authenticationHeaders) {
+    let subOrder = message.body.currentMessage.data
+    if (Array.isArray(subOrder)) {
+      subOrder = subOrder[0]
+    }
```

Comment:
This looks like a code smell to me, it is us that control the data in the `currentMessage` so why do we need to allow for 2 cases of `subOrder` being either a suborder or an array?